homeassistant/components/zwave/__init__.py

```diff
@@ -355,6 +358,16 @@ def _add_node_to_component():
             hass.data[DATA_DEVICES][entity.unique_id] = entity
             component.add_entities([entity])
 
+            # Update the created entity with the entry ID
```

Comment:
@balloob I'm a bit unsure about this bit. Would it be better to set the config entry in the `EntityComponent` above? There doesn't seem to be any way to do that though.

Reply:
I think that it would make more sense for Z-Wave to instantiate `EntityPlatform` directly and ignore `EntityComponent`, and then set the config entry on there.

Reply:
Done 😀